### PR TITLE
[v1.1 DE-NAS ASR] [ISSUE-184] DE-NAS ASR load pre-trained model

### PR DESCRIPTION
**Pull request by Peach-He**
### What changes were proposed in this pull request?

* add `load_pretrained_model` for DE-NAS ASR to load pre-trained model

### Why are the changes needed?

* add support of DE-NAS ASR to load pre-trained model
* for integration test with Pruner

### How was this patch tested?

* this patch test will be triggered in UT & CI/CD

### CODE REVIEW

**Comment by Peach-He** [comment]
@xuechendi DE-NAS ASR pre-work for Pruner. Enable ASR model builder to load pre-trained model and will use in Pruner integration.

File: e2eAIOK/DeNas/asr/model_builder_denas_asr.py [hunk 1]
```diff
@@ -51,4 +57,81 @@ def _init_model(self):
...
+    return model
+
+def load_pretrained_model(ckpt):
+    if not os.path.exists(ckpt):
+        raise RuntimeError(f"Can not find pre-trained model {ckpt}!")
+    logger.info(f"loading pretrained model at {ckpt}")
+
+    cfg = edict(supernet_config)
+    super_model = ModelBuilderASRDeNas(cfg)._init_model()
+    super_model_list = torch.nn.ModuleList([super_model["CNN"], super_model["Transformer"], super_model["seq_lin"], super_model["ctc_lin"]])
```
Comment (xuechendi) [inline e2eAIOK/DeNas/asr/model_builder_denas_asr.py:129]: why there is a duplicated function of line 62?
This is only for loading previous asr super net right?

Comment (Peach-He) [reply e2eAIOK/DeNas/asr/model_builder_denas_asr.py]: > why there is a duplicated function of line 62? 
removed duplicated load pre-trained model function

>This is only for loading previous asr super net right?
yes, added load pre-trained model function for ASR model builder

Comment (xuechendi) [reply e2eAIOK/DeNas/asr/model_builder_denas_asr.py]: @Peach-He ， where is `load_pretrained_model` being called?

Comment (Peach-He) [reply e2eAIOK/DeNas/asr/model_builder_denas_asr.py]: It is not called for now, just provides function here. `load_pretrained_model` will be callled in `search.py` to load pre-trained model after Pruner integrated in DE-NAS search.

File: e2eAIOK/DeNas/asr/model_builder_denas_asr.py [hunk 2]
```diff
@@ -51,4 +57,20 @@ def _init_model(self):
...
\ No newline at end of file
+        return model
+
+    def load_pretrained_model(self):
+        if not os.path.exists(self.cfg['ckpt']):
+            raise RuntimeError(f"Can not find pre-trained model {self.cfg['ckpt']}!")
+        print(f"loading pretrained model at {self.cfg['ckpt']}")
+
+        super_model = self._init_model()
+        super_model_list = torch.nn.ModuleList([super_model["CNN"], super_model["Transformer"], super_model["seq_lin"], super_model["ctc_lin"]])
```
Comment (xuechendi) [inline e2eAIOK/DeNas/asr/model_builder_denas_asr.py:68]: how did you define these supermodels here?

File: e2eAIOK/DeNas/asr/model_builder_denas_asr.py (hunk 1 above)
Comment (xuechendi) [reply e2eAIOK/DeNas/asr/model_builder_denas_asr.py]: @Peach-He , Oh, I thought you said this PR is to fix previous bugs for ASR supernet based codes. If load_pretrained_model is for pruner only, let's submit it later. We need to complete CI/CD and unittests covering before merge this part.

**Review by xuechendi — APPROVED** [review]
approved

**Pull request merged**